app/lib/primer/status/dsl.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Primer
+  # :nodoc:
+  module Status
+    # DSL to allow components to register their status.
+    #
+    # Example:
+    #
+    # class MyComponent < ViewComponent::Base
+    #   include Primer::Status::Dsl
+    #   status :beta
+    # end
+    module Dsl
+      extend ActiveSupport::Concern
+
+      STATUSES = {
+        alpha: :alpha,
+        beta: :beta,
+        stable: :stable,
+        deprecated: :deprecated
+      }.freeze
+
+      class UnknownStatusError < StandardError; end
+
+      included do
+        class_attribute :component_status, instance_writer: false, default: STATUSES[:alpha]
+      end
+
+      class_methods do
+        def status(status = nil)
```

Comment:
Is it typical to have this method be both the writer and reader (depending on if the arg is passed)?

Reply:
I borrowed this idea from the `view_component` framework. They do the same for SlotsV2 where `.slot_name` returns the slot and `.slot_name(xyz)` sets the slot

I thought it was a good way to not break the read API so I didn't have to change tests and dumps